src/basicio.cpp

```diff
@@ -1106,8 +1108,10 @@ namespace Exiv2 {
 
         if (need > size_) {
             if (need > sizeAlloced_) {
-                // Allocate in blocks of 32kB
-                long want = 32768 * (1 + need / 32768);
+                blockSize = 2*sizeAlloced_ ;
+                if ( blockSize > maxBlockSize ) blockSize = maxBlockSize ;
+                // Allocate in blocks
+                long want      = blockSize * (1 + need / blockSize );
                 data_ = (byte*)std::realloc(data_, want);
```

Comment:
Same as `malloc`, can return `NULL`.

Reply:
You're right.  I was so focused on the performance/blockSize matter that I didn't bother to think about malloc/realloc returning NULL.

I'm not so convinced about using std::vector.  MemIo is an in-memory file/stream in which the data is stored in a contiguous block of memory.  We could use std::vector to hold an array of bytes, however we probably need "blockSize" to request more bytes.  We could use std::vector to create an array of discontiguous blocks of memory.  This would avoid a single large contiguous memory block and complicate read() and write() to deal with block boundaries.  That design would avoid copying bytes in the file/stream as it grows.

When the bigtiff/multi-page tiff project moves forward, we could be working with medical imaging files of 10+Gbtyes from the folks at Imperial College.  It's unlikely that MemIo objects will be viable and we should re-instate the use of a temporary file on local storage in tiffimage.cpp/bigtiffimage.cpp

For the moment, I propose to:
1) Test/throw for NULL from malloc/realloc
2) retain the malloc/realloc contiguous block strategy